mlflow/pipelines/steps/transform.py

```diff
@@ -98,7 +113,7 @@ def get_identity_transformer():
             transformer_fn = getattr(
                 importlib.import_module(transformer_module_name), transformer_method_name
             )
-            transformer = transformer_fn()
+            transformer = _validate_user_code_output(transformer_fn)
```

Comment:
Prefer to call the `transformer_fn()` explicitly here so that `_validate_user_code_output` input argument is actually the output of the `transformer_fn`, instead of the function itself.

Reply:
I was thinking about following this theme similar to:
https://github.com/mlflow/mlflow/blob/6fde7b7230c7e1894a8c3beb9fd75ab996c3d640/mlflow/pipelines/steps/split.py#L112
and 
https://github.com/mlflow/mlflow/blob/8649fcff7b2a5aa32584e23e2e3ce26653b1865f/mlflow/pipelines/steps/ingest/datasets.py#L439

where we don't call the functions directly rather pass in the arguments. 

What do you think?